lib/matplotlib/colors.py

```diff
@@ -907,8 +907,11 @@ def process_value(value):
         if np.issubdtype(dtype, np.integer) or dtype.type is np.bool_:
             # bool_/int8/int16 -> float32; int32/int64 -> float64
             dtype = np.promote_types(dtype, np.float32)
+        # use np.asarray so data passed in as an ndarray subclass are
+        # interpreted as an ndarray. See issue #6622.
         result = np.ma.array(value, dtype=dtype, copy=True)
-        return result, is_scalar
+        result_data = np.asarray(result.data)
```

Comment:
I think that all you really need to do here is add the kwarg `subok=False` to the `np.ma.array()` line returning `result`.  Then `result` will be a plain masked array, and result.data will be a plain ndarray, so you will not need to extract and return it here.  The reason all this came up is that `np.ma.array` has a surprising default of `subok=True`, unlike `np.array`, which has the default `subok=False`.